programs/survival/hammer.sc

```diff
@@ -12,36 +27,41 @@ __get_block_range(position, face) ->
 );
 __harvestable(block) -> !(air(block) || (block == 'lava') || (block == 'water'));
 
+global_active = false;
 global_radius = 1;
 global_breakrange = null;
 
-__on_player_right_clicks_block(player, item_tuple, hand, block, face, hitvec) ->
-(
-	if (hand != 'mainhand' || !item_tuple || item_tuple:0 !='stone_pickaxe', return());
-	global_radius = (global_radius + 1)%3;
-	diameter = 2*global_radius+1;
-	print('hammer mode '+diameter+'x'+diameter);
-	if(global_radius,
-		for(__get_block_range(pos(block), face), 
-			particle('block '+block(_), _, 50, 0, 0.7)
+__on_player_right_clicks_block(player, item_tuple, hand, block, face, hitvec) -> (
+	if(global_active && hand == 'mainhand' && item_tuple && item_tuple:0 =='stone_pickaxe',
+		global_radius = (global_radius + 1)%3;
+		diameter = 2*global_radius+1;
+		display_title(player, 'actionbar', format('y hammer mode '+diameter+'x'+diameter));
+		if(global_radius,
+			for(__get_block_range(pos(block), face), 
+				particle('block '+block(_), _, 50, 0, 0.7)
+			)
 		)
-	)
+	);
 );
 
-__on_player_clicks_block(player, block, face) -> if (global_radius,
-	global_breakrange = null;
-	item_mainhand = player ~ 'holds';
-	if (!item_mainhand || !(item_mainhand:0 == 'stone_pickaxe'), return());
-	global_breakrange = __get_block_range(pos(block), face);
-	for(global_breakrange, particle('block '+block(_), _, 100, 0, 0.7))
+__on_player_clicks_block(player, block, face) -> (
+	if (global_radius && global_active,
+		global_breakrange = null;
+		
+		item_mainhand = player ~ 'holds';
+		if (!item_mainhand || !(item_mainhand:0 == 'stone_pickaxe'), return());
+		global_breakrange = __get_block_range(pos(block), face);
+		for(global_breakrange, particle('block '+block(_), _, 100, 0, 0.7))
+	);
 );
 
-__on_player_breaks_block(player, block) -> 
-(
+__on_player_breaks_block(player, block) -> (
 	block_range = global_breakrange;
 	global_breakrange = null;
-	if (!global_radius || !block_range, return());
-	item_mainhand = player ~ 'holds';
-	if(!item_mainhand || !(item_mainhand:0 == 'stone_pickaxe'), return());
-	for(block_range, harvest(player, _); signal_event('player_breaks_block', player, player, _))
```

Comment:
> You removed the event signal here, is that intended?

Yes, that was intentional
> Though idk if it does anything anymore since events no longer trigger inside an event handler (does that apply for signal_event?).

Wdym?

Reply:
Oh, that since a while ago (not sure what version) Carpet doesn't trigger events (idk if all or just some) during an event. It was done for a fix to modded changes being ignored while preventing event recursivity iirc.

> Yes, that was intentional
I'll merge this then